pkg/stanza/fileconsumer/file.go

```diff
@@ -58,6 +61,9 @@ func (m *Manager) Start(persister operator.Persister) error {
 			m.readerFactory.FromBeginning = true
 			m.tracker.LoadMetadata(offsets)
 		}
+		m.archive.SetStorageClient(persister)
+	} else if m.pollsToArchive > 0 {
+		return fmt.Errorf("archiving is not supported in memory, please use a storage extension")
```

Comment:
@djaglowski For now, I'm inclining to throw an error if archiving is enabled and no storage extension is used.